### ISSUE TITLE
When adding a Yamaha musiccast mediaplayer, all mediaplayers disappear

### ISSUE THREAD

**Issue by HA-TB303** [music-assistant/hass-music-assistant#421]
https://github.com/music-assistant/hass-music-assistant/discussions/409#discussioncomment-2976691

Music Assistant 2022.6.2
Home Assistant Core 2022.6.6
Home Assistant Supervisor 2022.05.3
Home Assistant OS 8.2
https://www.home-assistant.io/integrations/yamaha_musiccast/

Reproduction path: Add several (working) mediaplayers. Test and verify those work.
Then add a Musiccast mediaplayer and all mediaplayer that previously worked, are no longer visible.

I am more then willing (and hopefully able) to help and test. I have several Yamaha MusicCast devices.

**Comment by erkr** [comment]
Hi @HA-TB303 Some additional information could help Marcel. 

MA hides the source players by default, and normally publishes new onces (You can recognize them as their ID starts with mass_. 

So please tell us what players you added to MA, and which of them disappeared?
Are some of them group players?

Best Eric

**Comment by HA-TB303** [comment]
If I add my google mini, nest and home devices they all appear as aspected. If I then add 1 single Musiccast mediaplayer, they all dissapear.

**Comment by HA-TB303** [comment]
So this:
<img width="581" alt="image" src="https://user-images.githubusercontent.com/9609805/174442442-ab525e25-6ca6-4fa6-92f7-b2b27397c8f0.png">

Now looks like:
<img width="431" alt="image" src="https://user-images.githubusercontent.com/9609805/174442455-bebc3b8a-6e84-44f9-8640-baaba5c7df17.png">

If I add: 
<img width="606" alt="image" src="https://user-images.githubusercontent.com/9609805/174442501-0fb432c1-9e8e-4aba-9e94-6e7075b07d6d.png">

Then It looks like:
<img width="458" alt="image" src="https://user-images.githubusercontent.com/9609805/174442518-d5f0a19a-b0fc-46ef-8ba8-f64c9f270595.png">

**Comment by erkr** [comment]
Thanks for this additional information, that is definitely not what should happen, This helps finding the cause. 
Anything related in your logfiles?

**Comment by HA-TB303** [comment]
Not sure if it's related, but it appears right after adding the musiccast device:
```
Logger: homeassistant.components.media_player
Source: custom_components/mass/player_controls.py:153
Integration: Media player ([documentation](https://www.home-assistant.io/integrations/media_player), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+media_player%22))
First occurred: 4:27:14 PM (1 occurrences)
Last logged: 4:27:14 PM

Error adding entities for domain media_player with platform mass
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 382, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 809, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/config/custom_components/mass/media_player.py", line 163, in async_added_to_hass
    await super().async_added_to_hass()
  File "/config/custom_components/mass/entity.py", line 36, in async_added_to_hass
    await self.async_on_update()
  File "/config/custom_components/mass/media_player.py", line 254, in async_on_update
    self._attr_media_position = self.player.active_queue.elapsed_time
  File "/usr/local/lib/python3.9/site-packages/music_assistant/models/player_queue.py", line 125, in elapsed_time
    return self.player.elapsed_time
  File "/config/custom_components/mass/player_controls.py", line 153, in elapsed_time
    diff = (utcnow() - last_upd).seconds
TypeError: can't subtract offset-naive and offset-aware datetimes
```

```
Logger: homeassistant
Source: custom_components/mass/player_controls.py:153
Integration: Music Assistant ([documentation](https://github.com/music-assistant/hass-music-assistant), [issues](https://github.com/music-assistant/hass-music-assistant/issues))
First occurred: 4:27:14 PM (2 occurrences)
Last logged: 4:27:14 PM

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/mass/__init__.py", line 137, in on_mass_event
    data = event.data.to_dict()
  File "/usr/local/lib/python3.9/site-packages/music_assistant/models/player_queue.py", line 727, in to_dict
    "elapsed_time": int(self.elapsed_time),
  File "/usr/local/lib/python3.9/site-packages/music_assistant/models/player_queue.py", line 125, in elapsed_time
    return self.player.elapsed_time
  File "/config/custom_components/mass/player_controls.py", line 153, in elapsed_time
    diff = (utcnow() - last_upd).seconds
TypeError: can't subtract offset-naive and offset-aware datetimes
```

**Comment by marcelveldt** [comment]
This is issue in musiccast integration. Already reported in core repo

**Comment by HA-TB303** [comment]
Great!

**Comment by frank451a** [comment]
Just to add to this. I have setup MA with a mixture of media players including Yamaha during the initial config. All were listed. However after now playing one of my Spotify playlists on my Yamaha 3080, all players have disappeared as have all the Spotify albums/songs and playlists. The playlist is still playing on the Yamaha though.

**Comment by marcelveldt** [comment]
I see that the patch has been released in HA 2022.6.7 so this issue can be closed.
There however also seems to be a playback issue but let's focus on that in a separate issue report.
